crates/router/src/connector/paybox/transformers.rs

```diff
@@ -1004,9 +1108,125 @@ impl TryFrom<&PayboxRouterData<&types::PaymentsCompleteAuthorizeRouterData>> for
+    pub activity: String,
+
+    #[serde(rename = "SITE")]
+    pub site: Secret<String>,
+
+    #[serde(rename = "RANG")]
+    pub rank: Secret<String>,
+
+    #[serde(rename = "CLE")]
+    pub key: Secret<String>,
+
+    #[serde(rename = "DATEVAL")]
+    pub cc_exp_date: Secret<String>,
+
+    #[serde(rename = "REFABONNE")]
+    pub customer_id: Secret<String>,
+
+    #[serde(rename = "PORTEUR")]
+    pub carrier_id: Secret<String>,
+}
+
+impl
+    TryFrom<(
+        &PayboxRouterData<&types::PaymentsAuthorizeRouterData>,
+        CardMandateInfo,
+    )> for MandatePaymentRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        (item, card_mandate_info): (
+            &PayboxRouterData<&types::PaymentsAuthorizeRouterData>,
+            CardMandateInfo,
+        ),
+    ) -> Result<Self, Self::Error> {
+        let auth_data: PayboxAuthType =
+            PayboxAuthType::try_from(&item.router_data.connector_auth_type)
+                .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        let transaction_type = match item.router_data.request.capture_method {
+            Some(enums::CaptureMethod::Automatic) | None => {
+                Ok(MANDATE_AUTH_AND_CAPTURE_ONLY.to_string())
+            }
+            Some(enums::CaptureMethod::Manual) => Ok(MANDATE_AUTH_ONLY.to_string()),
+            _ => Err(errors::ConnectorError::CaptureMethodNotSupported),
+        }?;
+        let currency = diesel_models::enums::Currency::iso_4217(&item.router_data.request.currency)
+            .to_string();
+        let format_time = common_utils::date_time::format_date(
+            common_utils::date_time::now(),
+            DateFormat::DDMMYYYYHHmmss,
+        )
+        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Self {
+            date: format_time.clone(),
+            transaction_type,
+            paybox_request_number: get_paybox_request_number()?,
+            amount: item.router_data.request.minor_amount,
+            description_reference: item.router_data.connector_request_reference_id.clone(),
+            version: VERSION_PAYBOX.to_string(),
+            currency,
+            activity: RECURRING_ORIGIN.to_string(),
+            site: auth_data.site,
+            rank: auth_data.rang,
+            key: auth_data.cle,
+            customer_id: Secret::new(
+                item.router_data
+                    .request
+                    .get_connector_mandate_request_reference_id()?,
+            ),
+            carrier_id: Secret::new(item.router_data.request.get_connector_mandate_id()?),
+            cc_exp_date: get_card_expiry_month_year_2_digit(
+                card_mandate_info.card_exp_month.peek().clone(),
+                card_mandate_info.card_exp_year.peek().clone(),
+            )?,
+        })
+    }
+}
+
+fn get_card_expiry_month_year_2_digit(
+    card_exp_month: String,
+    card_exp_year: String,
```

Comment:
Can this function accept `Secret<String>` instead? We can call `peek()` within the function.

Reply:
changed